jetty-util/src/main/java/org/eclipse/jetty/util/security/Credential.java

```diff
@@ -70,10 +72,7 @@
      */
     public static Credential getCredential(String credential)
     {
-        if (credential.startsWith(Crypt.__TYPE)) return new Crypt(credential);
-        if (credential.startsWith(MD5.__TYPE)) return new MD5(credential);
-
-        return new Password(credential);
+        return CREDENTIAL_FACTORY.getCredential(credential);
```

Comment:
Why have a separate credential factory class?   Why not just put the behaviour here?